webapi/src/main/scala/org/knora/webapi/routing/BaseEndpoints.scala

```diff
@@ -6,30 +6,92 @@
+import dsp.errors.DuplicateValueException
+import dsp.errors.ForbiddenException
 import dsp.errors.NotFoundException
 import dsp.errors.RequestRejectedException
+import dsp.errors.ValidationException
+import org.knora.webapi.messages.StringFormatter
+import org.knora.webapi.messages.admin.responder.usersmessages.UserADM
+import org.knora.webapi.messages.admin.responder.usersmessages.UserIdentifierADM
+import org.knora.webapi.messages.v2.routing.authenticationmessages.KnoraCredentialsV2.KnoraPasswordCredentialsV2
 
-final case class BaseEndpoints() {
+final case class BaseEndpoints(authenticator: Authenticator, implicit val r: zio.Runtime[Any]) {
 
   private val defaultErrorOutputs: EndpointOutput.OneOf[RequestRejectedException, RequestRejectedException] =
     oneOf[RequestRejectedException](
       oneOfVariant[NotFoundException](statusCode(StatusCode.NotFound).and(jsonBody[NotFoundException])),
-      oneOfVariant[BadRequestException](statusCode(StatusCode.BadRequest).and(jsonBody[BadRequestException]))
+      oneOfVariant[BadRequestException](statusCode(StatusCode.BadRequest).and(jsonBody[BadRequestException])),
+      oneOfVariant[ValidationException](statusCode(StatusCode.BadRequest).and(jsonBody[ValidationException])),
+      oneOfVariant[DuplicateValueException](statusCode(StatusCode.BadRequest).and(jsonBody[DuplicateValueException]))
+    )
+
+  private val secureDefaultErrorOutputs: EndpointOutput.OneOf[RequestRejectedException, RequestRejectedException] =
+    oneOf[RequestRejectedException](
+      // default
+      oneOfVariant[NotFoundException](statusCode(StatusCode.NotFound).and(jsonBody[NotFoundException])),
+      oneOfVariant[BadRequestException](statusCode(StatusCode.BadRequest).and(jsonBody[BadRequestException])),
+      oneOfVariant[ValidationException](statusCode(StatusCode.BadRequest).and(jsonBody[ValidationException])),
+      oneOfVariant[DuplicateValueException](statusCode(StatusCode.BadRequest).and(jsonBody[DuplicateValueException])),
+      // plus security
+      oneOfVariant[BadCredentialsException](statusCode(StatusCode.Unauthorized).and(jsonBody[BadCredentialsException])),
+      oneOfVariant[ForbiddenException](statusCode(StatusCode.Forbidden).and(jsonBody[ForbiddenException]))
     )
 
   val publicEndpoint = endpoint.errorOut(defaultErrorOutputs)
+
+  val securedEndpoint = endpoint
+    .errorOut(secureDefaultErrorOutputs)
+    .securityIn(auth.bearer[Option[String]](WWWAuthenticateChallenge.bearer))
+    .securityIn(cookie[Option[String]](authenticator.calculateCookieName()))
+    .securityIn(auth.basic[Option[UsernamePassword]](WWWAuthenticateChallenge.basic("realm")))
+    .serverSecurityLogic {
+      case (Some(jwtToken), _, _) => authenticateJwt(jwtToken)
+      case (_, Some(cookie), _)   => authenticateJwt(cookie)
+      case (_, _, Some(basic))    => authenticateBasic(basic)
+      case _                      => Future.successful(Left(BadCredentialsException("No credentials provided.")))
+    }
```

Comment:
are you silently dropping support for authenticating with query parameters? If so, I applaud you :D

Reply:
Yes, this is not implemented for the admin/project endpoints. 
I would like to drop support for that in the long run entirely.
I guess we will have to figure out if it actually still in use.
Should be worth noting on the release notes.

Reply:
also, docs need to be updated

Reply:
Good point, updated the documentation in 9ed8a02d052682f3a320ad24f22c73bf7610ec43